### PR TITLE
refactor(sdk): create NilIfZero util

### PR DESCRIPTION
**Pull request by szymon-piechowicz-wandb**
Fixes WB-15188

# Description

Protocol buffers don't support null, so missing values are stored as zero. We often want to convert zero back to null. There was an existing `emptyAsNil` function for this. I generalized it to work for any type (instead of just string) and moved it to a reusable `pkg/utils` package.

# Test plan

Tested in https://github.com/wandb/wandb/pull/6296

### CODE REVIEW

File: nexus/pkg/utils/utils.go [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package utils
+
+func NilIfZero[T comparable](x T) *T {
```
Comment (dmitryduev) [inline nexus/pkg/utils/utils.go:3]: nice!

**Comment by szymon-piechowicz-wandb** [comment]
I know "utils" is an evil name that tends to become a catch-all. In this case though I think it's fitting. Or do you have better suggestions?

**Comment by kptkin** [comment]
> I know "utils" is an evil name that tends to become a catch-all. In this case though I think it's fitting. Or do you have better suggestions?

Maybe make it more concrete to the types of utility functions that would go into this package same as the current one

**Comment by dmitryduev** [comment]
> I know "utils" is an evil name that tends to become a catch-all. In this case though I think it's fitting. Or do you have better suggestions?

How about "transformations" or "datatypes"?

**Review by dmitryduev — APPROVED** [review]
👍🏻 Let's sort out the nits later, and get these PRs in!

**Pull request merged**